pygmo/docstrings.cpp

```diff
@@ -1938,4 +1938,155 @@ Runs the fast non dominated sorting algorithm on the input *points*
 )";
 }
 
+std::string hv_init1_docstring()
+{
+    return R"(__init__(pop)
+
+Constructor from population
+
+Args:
+    pop (``pygmo.population``): a pygmo.population
+
+Raises:
+    ValueError: if *pop* contains a single-objective or a constrained problem
+
+Examples:
+    >>> from pygmo import *
+    >>> pop = population(prob = zdt(id = 1), size = 20)
+    >>> hv = hypervolume(pop = pop)
+
+See also the docs of the C++ class :cpp:class:`pagmo::hypervolume`.
+
+)";
+}
+
+std::string hv_init2_docstring()
+{
+    return R"(__init__(points)
+
+Constructor from points
+
+Args:
+    points (``list`` of ``arrays``): the points
+
+Raises:
+    ValueError: if *points* is inconsistent
+
+Examples:
+    >>> from pygmo import *
+    >>> points = [[1,2],[0.5, 3],[0.1,3.1]]
+    >>> hv = hypervolume(points = points)
+
+See also the docs of the C++ class :cpp:class:`pagmo::hypervolume`.
+
+)";
+}
+
+std::string hv_compute_docstring()
+{
+    return R"(hypervolume.compute(ref_point, hv_algo = auto)
+
+Computes the hypervolume with the supplied algorithm. If no algorithm
+is supplied,  then an exact hypervolume algorithm is automatically selected
+specific for the point dimension.
+
+Args:
+    ref_point (``list`` of ``arrays``): the points
+    hv_algo (``pygmo.hv_algorithm``): hypervolume algorithm to be used
+
+Returns:
+    ``float``: the computed hypervolume assuming *ref_point* as reference point
+
+Raises:
+    ValueError: if *ref_point* is not dominated by the nadir point
+
+See also the docs of the C++ class :cpp:func:`pagmo::hypervolume::compute`.
+
+)";
+}
+
+std::string hv_contributions_docstring()
+{
+    return R"(hypervolume.contributions(ref_point, hv_algo = auto)
+
+This method returns the exclusive contribution to the hypervolume of every point.
+According to *hv_algo* this computation can be implemented optimally (as opposed to calling
+for :func:`~pygmo.hypervolume.exclusive` in a loop).
+
+Args:
+    ref_point (``list`` of ``arrays``): the points
+    hv_algo (``pygmo.hv_algorithm``): hypervolume algorithm to be used
+
+Returns:
+    ``Numpy array``: the contribution of all points to the hypervolume
+
+Raises:
+    ValueError: if *ref_point* is not suitable
+
+See also the docs of the C++ class :cpp:func:`pagmo::hypervolume::contributions`.
+
+)";
+}
+
+std::string hv_exclusive_docstring()
+{
+    return R"(hypervolume.exclusive(idx, ref_point, hv_algo = auto)
+
+Computes the exclusive contribution to the hypervolume of a particular point.
+
+Args:
+    idx (``int``): index of the point
+    ref_point (``list`` of ``arrays``): the reference points
+    hv_algo (``pygmo.hv_algorithm``): hypervolume algorithm to be used
+
+
+Returns:
+    ``Numpy array``: the contribution of all points to the hypervolume
+
+Raises:
+    ValueError: if *ref_point* is not suitable
```

Comment:
I'd prefer a list of cases in which ``ValueError`` is thrown, instead of multiple ``Raises`` clauses.